src/app/lists/dataview.tsx

```diff
@@ -0,0 +1,82 @@
+"use client";
+
+import Table from "@/components/Tables/Table";
+import supabase from "@/supabase";
+import React from "react";
+import { styled } from "styled-components";
+
+const TableDiv = styled.div`
+    margin: 20px;
+    padding: 20px;
+    background-color: ${(props) => props.theme.surfaceBackgroundColor};
+
+    border: solid 1px ${(props) => props.theme.surfaceForegroundColor};
+    border-radius: 1rem;
+`;
+
+const StyledTable = styled(Table)`
+    width: 100%;
+    background-color: transparent;
+`;
+
+type ListRow = { [headerKey: string]: string };
+
+const ListsDataView = async (): Promise<React.ReactElement> => {
+    // const rawData = (await supabase.from("lists").select("*")).data;
+    const rawData = [];
+    for (let i = 0; i < 100; i++) rawData.push({ item_name: `Item ${i}` });
+
+    console.log("fetched data");
+
+    const dataAndTooltips = rawData?.map((row) => {
```

Comment:
this is such a massive function I'd be tempted to extract to a different (named) function outside of the component